stl/inc/chrono

```diff
@@ -6040,58 +6052,47 @@ struct formatter<_CHRONO utc_time<_Duration>, _CharT> {
     }
 
 private:
-    _Chrono_formatter<_CharT, false> _Impl;
+    _Chrono_formatter<_CharT> _Impl{_STATICALLY_WIDEN(_CharT, "UTC")};
 };
 
 template <class _Duration, class _CharT>
 struct formatter<_CHRONO tai_time<_Duration>, _CharT> {
-    formatter() {
-        _Impl._Time_zone_abbreviation = _STATICALLY_WIDEN(_CharT, "TAI");
-    }
-
     auto parse(basic_format_parse_context<_CharT>& _Parse_ctx) {
         return _Impl.template _Parse<_CHRONO tai_time<_Duration>>(_Parse_ctx);
     }
 
     template <class _FormatContext>
     auto format(const _CHRONO tai_time<_Duration>& _Val, _FormatContext& _FormatCtx) {
-        const auto _Sys = _CHRONO sys_time<_Duration>{_Val.time_since_epoch()}
-                        - (_CHRONO sys_days{_CHRONO year{1970} / 1 / 1} - _CHRONO sys_days{_CHRONO year{1958} / 1 / 1});
+        using namespace chrono;
+        constexpr auto _Offset = sys_days{year{1970} / January / 1} - sys_days{year{1958} / January / 1};
+        const auto _Sys        = sys_time<_Duration>{_Val.time_since_epoch()} - _Offset;

```

Comment:
Small optimization: `_Offset` is going to be converted to `common_type_t<_Duration, days>`. My experiment with [Godbolt](https://godbolt.org/z/7cn3o5vnM) suggests that this is inlined with `/O2`, but at least for debug you can avoid a call to `duration_cast` by converting it at compile time (notice that the `operator+` with different `duration` specializations has an additional constructor call). Same for `gps_time` specialization.
```suggestion
        constexpr common_type_t<_Duration, days> _Offset{sys_days{year{1970} / January / 1} - sys_days{year{1958} / January / 1}};
        const auto _Sys = sys_time<_Duration>{_Val.time_since_epoch()} - _Offset;
```

Reply:
Thanks, this is a great catch! I'll add a named typedef with a comment explaining the purpose.